### PR TITLE
Disallow assignments to imported exports

### LINKED ISSUE #2456: Import namespace exports should be immutable
We should not allow mutating a property of a namespace import:
```ts
// Library
export var x = 1;

// Consumer
import * as stuff from 'Library';
stuff.x = 0;
```

In ES6, these bindings are immutable.

### PR DESCRIPTION
**Pull request by vladima**
Fixes #2456

### CODE REVIEW

File: src/compiler/checker.ts [hunk 1]
```diff
@@ -7254,14 +7253,37 @@ module ts {
                 }
             }
 
+            function isImportedNameFromExternalModule(n: Node): boolean {
```
Comment (yuit) [inline src/compiler/checker.ts]: This is already an inner function so it should be able to refer to the node from the outer function. Why you have to pass it in as function argument?

Comment (vladima) [inline src/compiler/checker.ts]: it is both the question of style (use explicit inputs\outputs for function) and potential optimization opportunities (in theory nested function that does not any captured data can be lifted to a higher scope)

**Comment by yuit** [comment]
:+1:

File: tests/cases/compiler/externalModuleImmutableBindings.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+// @Filename: f2.ts
+import * as stuff from 'f1';
+
+var n = 'baz';
+
+stuff.x = 0;
+stuff['x'] = 1;
+stuff.blah = 2;
+stuff[n] = 3;
+
```
Comment (mhegazy) [inline tests/cases/compiler/externalModuleImmutableBindings.ts]: `import x = stuff;`

**Comment by mhegazy** [comment]
:+1:

**Pull request merged**